packages/search/src/ago/format-item.ts

```diff
@@ -0,0 +1,64 @@
+// helper functions to make items look more like datasets
+//
+// item queries will look like this ->
+// {
+//   data: [{
+//     id: item-id,
+//     type: 'item',
+//     attributes: {
+//       // will have attribute 'parity' with datasets. Many of these values will be null.
+//     },
+//     meta: { highlights: {} }
+//     links: {
+//       // computed links for the item result. These can be used throughout the app when linking to
+//       // AGO, APIs, etc
+//     }
+//   }],
+//   meta: {
+//     aggs: {
+//       // built by another query against AGO API and then formatted in the same format as the V3 API
+//     },
+//     stats: {
+//       count:,
+//       totalCount:,
+//     }
+//   }
+// }
+import { hubTypeLookup } from "../common/hub-type-map";
+import { calcHighlights } from "../common/highlights";
+import { IItem } from "@esri/arcgis-rest-common-types";
+
+export function formatItem(item: IItem, query: string) {
+  const formattedItem: any = {
+    id: item.id,
+    type: "item",
+    attributes: formatItemAttributes(item)
+  };
+  if (query) {
+    if (!item.meta) {
+      // create highlights since AGO deos not return them
+      formattedItem.meta = {};
+      formattedItem.meta.highlights = highlights(item, query);
+    }
+  }
+  return formattedItem;
+}
+
+function formatItemAttributes(item: IItem) {
+  const hubTypes = hubTypeLookup(item.type);
+  return Object.assign({}, item, {
+    /* computed or null attributes so items & datasets look the same */
+    name: item.title,
+    searchDescription: item.description,
+    hubType: hubTypes[0] || "Other"
+  });
+}
+
+function highlights(item: IItem, query: string) {
```

Comment:
if you doc the return type, it'll show up in the API reference automatically.
```ts
function highlights(item: IItem, query: string):{ name: string, searchDescription: string } {
```

Reply:
sweet. I'll add the niceties once the basics are in place